cpp/src/arrow/array/builder_nested.h

```diff
@@ -771,6 +771,13 @@ class ARROW_EXPORT StructBuilder : public ArrayBuilder {
     return Status::OK();
   }
 
+  /// This method is "unsafe" because it does not update the null bitmap
+  /// It doesn't check the capacity. Caller is responsible for calling Reserve()
+  /// beforehand.
+  void UnsafeAppend(){
+    ++length_;
```

Comment:
```suggestion
    UnsafeAppendToBitmap(is_valid);
```

Could you use `UnsafeAppend(is_valid)` in `Append()` instead of `UnsafeAppendToBitmap(is_valid)`?